junit/src/main/java/org/kaazing/k3po/junit/rules/ScriptRunner.java

```diff
@@ -214,6 +214,7 @@ public void notifyBarrier(final String barrierName) throws InterruptedException
 
 
     public void dispose() throws Exception {
+        latch.awaitFinished();
```

Comment:
What are you trying to fix here, and what condition or sequence is causing it to be an issue?

I'm concerned that the tests might get into a state where it blocks here forever, specifically, SpecificationStatement evaluate has a lot of catching of exceptions on latch and that would trigger a finally that calls this.  It's hard to reason if whether in all these cases that the scriptFuture Thread will still be in a good state AND latch finished will eventually be triggered.